lib/internal/worker/io.js

```diff
@@ -127,11 +127,15 @@ ObjectDefineProperties(MessageEvent.prototype, {
   },
 });
 
+const originalCreateEvent = EventTarget.prototype[kCreateEvent];
 ObjectDefineProperty(
   MessagePort.prototype,
   kCreateEvent,
   {
     value: function(data, type) {
+      if (type !== 'message' && type !== 'messageerror') {
```

Comment:
@addaleax  Is this what you meant in https://github.com/nodejs/node/issues/35835#issuecomment-717959806 ?

Reply:
Yes :)